### DISCUSSION TITLE
Gesamtzählerstände haben keine Einheit

### DISCUSSION THREAD

**Discussion by horstvogel** [marq24/ha-evcc#7] [General]
Moin,
Gesamtzählerstände haben keine Einheit
ist das "Absicht", da diese dann ja eventuell bei einem Neustart von EVCC dann Null wären und daher dann z.B. das Energie Dashboard einen Anstieg von Null auf XX verzeichnen würde? 

Vielen Dank!

**Comment by marq24** [comment]
hast Du eine EntityID für mich - da ich aktuell nicht verstehe, was Du mit den _Gesamtzählerständen_ meinst - oder ggf einen Screenshot?!

**Reply by MrMattn** [reply]
hi
Horst antwortet sicher später noch.
Hab mit ihm im EVCC SlackChat geschrieben. Er meinte die Werte kommen ohne Einheit an und so könnte man sie zb nicht in das Energiedashboard einpflegen.
Mir selber ging es darum:
alles super funktioniert.
EVCC läuft in einem Container auf proxmox. 
Dein EVCC Addon hab ich in HA eingefügt. Hab direkt alle Werte bekommen. Aber wenn ich nun die Werte vom Stromzähler, der vor der Wallbox sitzt in HA in die PowerFlowCardplus einfüge, dann sehe ich manchmal kleine Stromverbräuche die nicht da sind. Selbst wenn das Auto nicht angeschlossen ist.

![Bildschirmfoto 2024-06-16 um 21 12 48 (1)](https://github.com/marq24/ha-evcc/assets/50632820/a1d89fc7-6c91-42d2-b019-9e792b7bcb3f)
![Bildschirmfoto 2024-06-16 um 21 11 06 (1)](https://github.com/marq24/ha-evcc/assets/50632820/507fa1ae-bd1d-4fa4-995d-bdbb5ebde0e2)

**Reply by marq24** [reply]
Ich will mich ja nicht "dümmer" anstellen, als ich bin - aber leider verstehe ich auch Deine Frage nicht :-/

> ...Aber wenn ich nun die Werte vom Stromzähler, der vor der Wallbox sitzt in HA in die PowerFlowCardplus einfüge, dann sehe ich manchmal kleine Stromverbräuche die nicht da sind. Selbst wenn das Auto nicht angeschlossen ist.

Also wenn Du einen Hardware-Sensor (z.B. Shelly) vor Deiner Wallbox hast - und dieser Sensor dann in HA (für Dich nicht plausible) Werte anzeigt, was könnte ich hier mit der evcc Integration daran ändern?

**Reply by UDicke** [reply]
`was könnte ich hier mit der evcc Integration daran ändern?`
Genau diese Frage habe ich mir auch gestellt...
und mir dann so beantwortet: **_NIX_** 😊

**Comment by horstvogel** [comment]
Ist dann Einheitenlos 



sensor.evcc_garage_rechts_charge_total_import




![image](https://github.com/marq24/ha-evcc/assets/25108737/49113de6-5abf-4555-b617-686dcc773e1a)


![image](https://github.com/marq24/ha-evcc/assets/25108737/bd48357a-ea90-4b73-9dc2-e4f8435a4416)

**Comment by horstvogel** [comment]
Um den Zählerstand dann für das Energie Dashboard nutzen zu können, müsste dann gewährleistet werden, dass der Zählerstand dann nicht auf Null geht. Für einen Energiezähler in der Config.yaml mache ich das so. 


`template:
  - sensor:
      - name: "Zählerstand"
        unit_of_measurement: kWh
        state_class: total_increasing
        device_class: energy
        state: >
          {% if states('Zaehlerwallbox') == 'unavailable' or states('Zaehlerwallbox') < '1' %}
            {{ states('sensor.zahlerstand') }}
          {% else %}
            {{ states('Zaehlerwallbox') }}
          {% endif %}`

**Comment by MrMattn** [comment]
interessanterweise wird bei marq24 alles mit Werten angezeigt?

![Bildschirmfoto 2024-06-18 um 19 27 47](https://github.com/marq24/ha-evcc/assets/50632820/3b2bbf95-26ac-4f3c-8f58-f577fff1f176)

**Comment by MrMattn** [comment]
by the way...bei mir auch Horst.
![Bildschirmfoto 2024-06-18 um 19 30 20](https://github.com/marq24/ha-evcc/assets/50632820/dbd99eca-7157-4ba1-b362-d77edb64bcf6)
![Bildschirmfoto 2024-06-18 um 19 29 38](https://github.com/marq24/ha-evcc/assets/50632820/8cdbd458-fbc3-4af7-a577-ce8accd462da)

**Comment by marq24** [comment]
@horstvogel - Danke! in der Tat, da "fehlt" was... update kommt - mit kWh oder Wh muss ich noch testen, was der original Wert der API ist, der zurückgeleifert wird - also ob das schon kWh sind - oder die API Wh liefert und ich dass noch in kWh umrechnen muss...

**Reply by horstvogel** [reply]
Moin;
bitte auch schauen, ob der Zustand unplausibel dann mit abgebildet werden kann. Sonst ist das ja für das Energie Dashboard so direkt nicht brauchbar, wenn dann der Wert wieder auf 0 und nicht erreichbar geht. 

Danke!

**Reply by marq24** [reply]
Sollte der Wert 'Netzbezug' denn nicht so wie ein Stromzähler funktionieren? Also der Wert dürfte sich über die Zeit nur größer werden? - Wenn der Wert auf 0 zurückfällt, dann wäre das aus meiner Sicht ein BUG in der API von evcc... [im 2024.6.2 update ist der Wert auf jeden fall auch als ''TOTAL_INCREASING" deklariert]

**Reply by horstvogel** [reply]
Neustart von EVCC = Kein Zustand...= kein Bug.

**Reply by horstvogel** [reply]
Wallbox ist mal nicht erreichbar...so habe ich gedacht, vielleicht liege ich ja falsch. 
Bei meinen Tasmota Dinger habe ich das bisher immer so gemacht.

**Reply by marq24** [reply]
> Neustart von EVCC = Kein Zustand...= kein Bug.

also, WENN das ein "Total-count" Zähler ist (wie gesagt ich setzte derzeit kein evcc bei mir ein), dann ist meine Erwartungshaltung, dass evcc den Wert bei einem Neustart des Systems den Wert aus der internen DB recovered... "Kein Zustand" ist m.E. nur dann zulässig, wenn der Zähler "neu" ist - und auch dann wäre mein Erwartungswert 0... 

Gleiches gibt m.E. für eine nicht erreichbare Wallbox (weil temp offline) - die Zahl/Wert kommt ja eben nicht aus der Wallbox - sondern ist ein Wert aus der evcc DB (also so zumindest meine Vermutung)...

**Reply by horstvogel** [reply]
Du hast das ja schon eingepflegt, die Werte kommen jetzt in kWh an. 

die Wallboxen stehen leider nicht in im Energiedashboard zur Verfügung. Nur die anderen Werte für die kWh des Ladevorganges etc. tauchen dort auf. 



![image](https://github.com/marq24/ha-evcc/assets/25108737/6bbc516b-0ee0-4d98-a55f-6f1ad9e2cf37)



![image](https://github.com/marq24/ha-evcc/assets/25108737/b73c6e37-b17b-4914-957d-7975a192ad03)

**Reply by marq24** [reply]
Kannst Du bitte etwas bei Dir lokal ausprobieren?

Das File `custom_components/evcc_intg/const.py` edditieren und die `Zeile 541` von

`        state_class=SensorStateClass.TOTAL_INCREASING,`

in

`        state_class=SensorStateClass.MEASUREMENT,`

ändern... und anschließend deine HA Instanz einmal durchstarten - taucht dann diese Entity in der Dashboard Auswahl auf?

**Reply by horstvogel** [reply]
Mache ich , aber das wäre ja schräg, fürs Dashboard müssten das doch Total Increasing Werte sein

**Reply by marq24** [reply]
Ich habe sonst KEINE Idee, was der Unterschied des Sensors gegenüber den Anderen (im EnergyDashboardZurAuswahlStehenden) sein sollte?!

**Comment by horstvogel** [comment]
Hat nichts geändert, ich starte jetzt den gesamten Server einmal neu, ist HA OS, also den Rechner starte ich ganz neu.

**Reply by horstvogel** [reply]
Nach einem Server Neustart 


![image](https://github.com/marq24/ha-evcc/assets/25108737/e4d52a21-ea24-45f4-8e9d-299638906db7)

**Reply by horstvogel** [reply]
Also auch keine Anzeige der Werte. 

ich deinstalliere einmal die komplette Integration und starte neu....

**Reply by horstvogel** [reply]
So komplette Neuinstallation von der Integration.... 
 und dann auch einmal wieder den Eintrag in der custom_components/evcc_intg/const.py   geändert. 
 
 Ich Werte tauchen noch immer nicht für das Energiedashboard auf.

**Reply by horstvogel** [reply]
Vielleicht hat ja jemand eine jungfräuliche Installation

**Comment by horstvogel** [comment]
So gerade bei einem Kumpel ausprobiert, der hat allerdings nur eine Wallbox, da hat es dann gleich funktioniert. 
Sprich der hatte noch nicht Deine Integration am Laufen.

Allerdings hat er nur eine Wallbox und nicht wie bei uns in der Firma 6 Wallboxen, könnte das noch ein Punkt sein?

**Reply by RenatusRo** [reply]
mal vollkommen unabhaengig von der integration und evcc: ich hab' hier +/- 50 shellies in HA. Und zwei davon sind nie ins energiedashboard gekommen. mit allen ueblichen tricks.
das gleiche mit template sensoren: x stueck gehen, 2 nicht :-(

**Comment by horstvogel** [comment]
Nach der Deinstallation sind die Entitäten nicht mehr in HA sichtbar.  die ganzen sensoren Finde ich aber noch unter den Statistiken. Ich gehe davon aus, dass das das Problem ist.

**Reply by horstvogel** [reply]
So ich habe die Statistiken gelöscht, HA Neustart ausgeführt und trotzdem werden noch Daten von heute morgen angezeigt, dass bedeutet das hier noch alte Daten vorliegen und daher vermutlich das alles durcheinander ist. 
Wobei wo kommen die her, wenn ich die gelöscht habe....

Nun kann ich allerdings die "Zähler" auch im Energiedashboard finden

**Reply by horstvogel** [reply]
Fazit: bei vorhandener Installation, bitte die Integration deinstallieren, unter Statistiken alle Werte löschen, Neustart und dann wieder installieren...so hat das bei mir dann abschließend funktioniert.

**Reply by UDicke** [reply]
Genau das ist das Problem gewesen (unterschiedliche Einheiten bei demselben Entity) . Ich habe das vermutet, weil ich sowas auch schon mal anderweitig hatte. Irgendwo in HA wurde mir da angeboten die Einheiten automatisch anzupassen (oder musste ich auch die Statistik löschen????), ich weiss es nicht mehr☹. Bevor ich das schreibe wollte ich natürlich testen, bekomme aber die neue Version nicht installiert. Ich habe die Integration schon gelöscht und nach HA Neustart wieder hinzugefügt. Trotzdem bleibt die bei mir bei 2024.6.1. Hat jemand eine Idee was ich machen kann?
![grafik](https://github.com/marq24/ha-evcc/assets/161384698/7b6f925f-7459-46fd-8324-c53687bdc07a)

**Reply by UDicke** [reply]
Ok, ich habe jetzt einfach das Update bei mir von Hand in die 3 Dateien eingepflegt. Jetzt sehe ich auch die Einheit kWh neben sensor.evcc_garage_rechts_charge_total_import. Im Energiedashboard wird er noch nicht angeboten. Mal sehen ob ich noch einen anderen Weg finde ohne die Statistik zu löschen...

**Reply by UDicke** [reply]
Ich habe die Lösung gefunden! 😊 Unter "Entwicklerwerkzeuge>Statistik" wird das geänderte Entity so angezeigt:
![Screenshot 2024-06-19 211558](https://github.com/marq24/ha-evcc/assets/161384698/f900e854-da01-42d8-b616-ff6f5bb137ca)

Beim Klicken auf "Problem bei..." wird angeboten die historischen Werte auf die neue Einheit (kWh) zu konvertieren oder die Statistik zu löschen. Ich habe mich für "konvertieren" entschieden und schon wird das besagte Entity im Energiedashboard angeboten.
![grafik](https://github.com/marq24/ha-evcc/assets/161384698/27c1cc56-9b9e-42ba-8fc4-e85df27cb0fc)

...und der Verlauf wird auch aus der Vergangenheit angezeigt.
![grafik](https://github.com/marq24/ha-evcc/assets/161384698/f56b4167-aa6c-4829-8c88-a91f74a33405)
Die Werte mit "0" zwischen 23:00 Uhr und 6:30 Uhr schiebe ich darauf, dass bei uns das W-Lan in diesem Zeitraum automatisch ausgeschaltet wird. @marq24 heisst für mich aber auch, dass die Werte nicht aus der evcc-DB kommen wenn sie nicht verfügbar sind, oder ist das eine abwegige Annahme?

**Reply by marq24** [reply]
naja - also wenn kein WLAN, dann kann vermutlich auch HA nicht mit evcc sprechen - oder aber eben evcc liefert "0", wenn keine Kommunikation mit der WB möglich ist... Das ist meiner Einschätzung nach aber alles ganz ganz weit von der Integration weg... Also ohne ein "stabiles" Hausnetz auch keine stabile HomeAutomation.

**Reply by UDicke** [reply]
Nachdem ich das wiedergefunden habe wollte ich ja auch nur schreiben, wie man das Problem mit der Statistik bei geänderter Einheit bei einem Entity auch lösen kann.

@marq24 mit Deiner Integration bin ich voll zufrieden so wie sie ist! 🥇 

Mit der Stabilität von unserem Hausnetz auch, W-Lan wird halt nachts ausgeschaltet und alles nötige funktioniert trotzdem.
evcc läuft als Add-On in HA und braucht kein W-Lan, nur die eine "Wallbox" (Steckdose mit Ladegerät über Shelly gemessen und geschaltet) kann sich in der Zeit nicht mitteilen. 😊
Im Energiedashbord hatte ich unsere Ladegeräte schon vor der Installation der evcc Intergration über andere Entities...
![grafik](https://github.com/marq24/ha-evcc/assets/161384698/68343f93-8237-4050-b792-9fdaf2fadad9)

**Comment by horstvogel** [comment]
So Werte tauchen auf im Energie Dashboard auf..
Vielen Dank! 

Wobei ich hatte das ja vorher schon anderes gelöst, über Mqtt; so ist das aber schon einfacher.

**Reply by MrMattn** [reply]
Also gibt es jetzt ein Update von dem addon? (Sehe nämlich nichts) oder was muss ich nun machen, damit beim Zähler auch kWh steht?

**Reply by marq24** [reply]
https://github.com/marq24/ha-evcc/releases/tag/2024.6.2

**Reply by MrMattn** [reply]
sorry aber meine github und programmier Künste sind eher mangelhaft :-D
hab jetzt die schöne zip Datei mit vielen Dateien runter geladen. was mache ich damit?

**Reply by UDicke** [reply]
@MrMattn wenn Du das Update möchtest und alleine nicht klar kommst kannst Du Dich gerne bei mir melden. Weil das Update leider nicht automatisch angeboten wird, habe ich das "von Hand" installiert. Siehe hier https://github.com/marq24/ha-evcc/discussions/7#discussioncomment-9818233
Es sind nur ein paar wenige Änderungen in insgesamt 3 Dateien. Siehe hier: https://github.com/marq24/ha-evcc/commit/7a719754da4703b44c09ea2d03dea8ef322b3e90

**Reply by MrMattn** [reply]
Wie kann ich mich denn bei dir melden außer hier :-)

**Reply by UDicke** [reply]
Inzwischen wird das Update bei mir in HACS angeboten...
![grafik](https://github.com/marq24/ha-evcc/assets/161384698/d0f0a6a5-78b3-46f8-b9d5-8b9ed8c0f566)
und hat auch funktioniert...😊
Ansonsten: u.dicke@gmx.de

**Reply by MrMattn** [reply]
Super hat geklappt!!

**Comment by horstvogel** [comment]
Einfach unter HACS erneut herunterladen und dann sollte die neuste Version installiert werden. 
Wichtig ist natürlich das Dein HACS "funktioniert" Oder wie hast Du die Integration installiert. 


![image](https://github.com/marq24/ha-evcc/assets/25108737/1423dcba-7042-475e-86d3-197bb1507674)

**Comment by horstvogel** [comment]
Moin, 

wie ich bereits vermutet hatte, falls mal keine Werte aus EVCC kommen oder halt eine Null. Das ist Chaos im Energiedashboard.

Der Werte war warum auch immer Null, vielleicht Kommunikation zur Wallbox ausgefallen.... und dann war 2x ein Verbrauch von über 2500 kWh. Was dann dem Gesamtzählerstand der Wallbox entspricht. 


![image](https://github.com/marq24/ha-evcc/assets/25108737/2dc9e8f3-b2d0-45a9-801f-b3d009f5b3bd)


![image](https://github.com/marq24/ha-evcc/assets/25108737/e4c1c5a8-b2e0-4ba9-a4eb-97c919c06543)

**Reply by horstvogel** [reply]
Daher hatte ich weiter oben schon beschrieben, dass ich das für die Daten die via MQTT dann reinkommen, so "vermeide".
Vielleicht kannst Du das ja auch in die Integration einbauen. Sonst wird das vermutlich mal hin und wieder dann "Chaos" geben


Um den Zählerstand dann für das Energie Dashboard nutzen zu können, müsste dann gewährleistet werden, dass der Zählerstand dann nicht auf Null geht. Für einen Energiezähler in der Config.yaml mache ich das so.

`template:

sensor:
name: "Zählerstand"
unit_of_measurement: kWh
state_class: total_increasing
device_class: energy
state: >
{% if states('Zaehlerwallbox') == 'unavailable' or states('Zaehlerwallbox') < '1' %}
{{ states('sensor.zahlerstand') }}
{% else %}
{{ states('Zaehlerwallbox') }}
{% endif %}`

**Comment by marq24** [comment]
Moin!

mit dem von Dir geposteten template, kannst Du doch auch den Sensor der Integration überschreiben? [also mit dem passenden Schnipsel an der Stelle `states('Zählerwallbox')` ]...

ich möchte nur sehr ungern innerhalb der integration von "Hand" dem User nicht plausible Werte wegfiltern - die Integration liefert das was die API einem zurückgibt... Denn das ist meiner bisherigen Erfahrung nach ein Fass ohne Boden... [vor allem wenn man die ganzen Usercases wie Neu-Installation (Wallbox, evcc, HA, HA-Integration) alle richtig regeln möchte...

Hier ein Screenshot meiner Kurve - hier siehst Du, das zumindest bei mir der Wert "nie" auf 0 runter fällt... Auch wenn ich evcc mal neu-starte (zb. wegen neuer Konfiguration)... dann ist da eben gar kein Wert (für die Zeit)... [und ja erst seit letztem Freitag hab ich auch endlich mein Auto bekommen... ]

![image](https://github.com/marq24/ha-evcc/assets/8012359/7f063c18-7edd-4b3a-9a40-6994a9275ae3)

**Comment by marq24** [comment]
@horstvogel  @UDicke -> check the latest release :-D

https://github.com/marq24/ha-evcc/releases/tag/2024.6.5

**Reply by MrMattn** [reply]
was macht man damit? (sorry bin nicht so der Programmierer)

**Reply by marq24** [reply]
es gibt ein Update der Integration (via HACS)

**Reply by MrMattn** [reply]
behebt das Update auch diese "Fehler"?
die 2 großen sind normale Ladungen. Ansonsten hängt das Auto gar nicht an. die letzten Tage war es recht ruhig.
Heute ist es ganz extrem

![Bildschirmfoto 2024-06-30 um 14 31 38](https://github.com/marq24/ha-evcc/assets/50632820/9bf15a4d-b396-45fc-b328-d69bb27ca91e)

![Bildschirmfoto 2024-06-30 um 14 32 47](https://github.com/marq24/ha-evcc/assets/50632820/53b932e0-e7db-46b8-b860-5a4b7d2d8c4f)
![Bildschirmfoto 2024-06-30 um 14 33 10](https://github.com/marq24/ha-evcc/assets/50632820/2c591cfd-d6b4-46b8-a68d-d97bb9b90da0)

**Reply by marq24** [reply]
ich kann Dir nicht folgen...

**Reply by MrMattn** [reply]
naja die HA EVCC Integration bekommt die Werte vom Stromzähler der vor der Wallbox hängt.
Und obwohl das Auto nicht angeschlossen ist, spuckt der Stromzähler manchmal Stromverbräuche aus, die eigentlich nicht existieren!? Diese sind mir früher wohl nie aufgefallen, hier werden sie aber visualisiert.

**Reply by horstvogel** [reply]
> @horstvogel @UDicke -> check the latest release :-D
> 
> https://github.com/marq24/ha-evcc/releases/tag/2024.6.5

Sieht gut aus, die Ausreißer habe ich die letzten Tage nicht mehr gehabt. 
Danke!
